src/fsharp/fsc.fs

```diff
@@ -1761,6 +1762,39 @@ let main0(ctok, argv, referenceResolver, bannerAlreadyPrinted, exiter:Exiter, er
         with e -> 
             errorRecoveryNoRange e
             exiter.Exit 1
+    
+    let inputs =
+        // Deduplicate module names
+        let seen = Dictionary<_,Set<_>>()
```

Comment:
It's unclear what type this `seen` thing has. Could you add a comment or, at least, types?